dojo/product/signals.py

```diff
@@ -1,77 +1,38 @@
-    if action in ["post_add", "post_remove"]:
-        if inherit_product_tags(instance):
-            tag_list = [tag.name for tag in instance.tags.all()]
-            if propagate_inheritance(instance, tag_list=tag_list):
-                instance.inherit_tags(tag_list)
-
-
-@receiver(signals.post_save, sender=Endpoint)
-@receiver(signals.post_save, sender=Engagement)
-@receiver(signals.post_save, sender=Test)
-@receiver(signals.post_save, sender=Finding)
-def inherit_tags_on_instance(sender, instance, created, **kwargs):
-    if inherit_product_tags(instance):
-        tag_list = instance._tags_tagulous.get_tag_list()
-        if propagate_inheritance(instance, tag_list=tag_list):
-            instance.inherit_tags(tag_list)
-
-
-def propagate_inheritance(instance, tag_list=[]):
-    # Get the expected product tags
-    product_inherited_tags = [tag.name for tag in get_product(instance).tags.all()]
-    existing_inherited_tags = [tag.name for tag in instance.inherited_tags.all()]
-    # Check if product tags already matches inherited tags
-    product_tags_equals_inherited_tags = product_inherited_tags == existing_inherited_tags
-    # Check if product tags have already been inherited
-    tags_have_already_been_inherited = set(product_inherited_tags) <= set(tag_list)
-    return not (product_tags_equals_inherited_tags and tags_have_already_been_inherited)
-
-
-def inherit_product_tags(instance) -> bool:
-    product = get_product(instance)
-    # Save a read in the db
-    if product and product.enable_product_tag_inheritance:
-        return True
-
-    return get_system_setting('enable_product_tag_inheritance')
-
-
-def get_product(instance):
-    if isinstance(instance, Product):
-        return instance
-    if isinstance(instance, Endpoint):
-        return instance.product
-    if isinstance(instance, Engagement):
-        return instance.product
-    if isinstance(instance, Test):
-        return instance.engagement.product
-    if isinstance(instance, Finding):
-        return instance.test.engagement.product
+from django.db.models.signals import post_save, post_delete
+from django.utils.translation import gettext as _
+from django.urls import reverse
+
+from dojo.models import Product
+from dojo.notifications.helper import create_notification
+
+
+@receiver(post_save, sender=Product)
+def product_post_save(sender, instance, created, **kwargs):
+    if created:
+        create_notification(event='product_added',
+                            title=instance.name,
+                            product=instance,
+                            url=reverse('view_product', args=(instance.id,)))
+
+
+@receiver(post_delete, sender=Product)
+def product_post_delete(sender, instance, **kwargs):
+    if settings.ENABLE_AUDITLOG:
+        le = LogEntry.objects.get(
+                action=LogEntry.Action.DELETE,
+                content_type=ContentType.objects.get(app_label='dojo', model='product'),
+                object_id=instance.id
```

Comment:
Username is properly populated here when deleting via the UI (`TestProduct`), but not the API (`test1`)...

![Screenshot 2024-04-09 at 20 31 54](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/f05e1d2a-aa1a-4d74-97ee-3be5dd50d44f)